pkg/plugins/policies/meshpassthrough/api/v1alpha1/validator.go

```diff
@@ -0,0 +1,88 @@
+package v1alpha1
+
+import (
+	"fmt"
+	"math"
+	"regexp"
+	"slices"
+	"strings"
+
+	"github.com/asaskevich/govalidator"
+
+	common_api "github.com/kumahq/kuma/api/common/v1alpha1"
+	"github.com/kumahq/kuma/pkg/core/resources/apis/mesh"
+	"github.com/kumahq/kuma/pkg/core/validators"
+	"github.com/kumahq/kuma/pkg/util/pointer"
+)
+
+var (
+	allMatchProtocols                = []string{string(TcpProtocol), string(TlsProtocol), string(GrpcProtocol), string(HttpProtocol), string(Http2Protocol)}
+	notAllowedProtocolsOnTheSamePort = []ProtocolType{GrpcProtocol, HttpProtocol, Http2Protocol}
+	wildcardPartialPrefixPattern     = regexp.MustCompile(`^\*[^\.]+`)
+)
+
+func (r *MeshPassthroughResource) validate() error {
+	var verr validators.ValidationError
+	path := validators.RootedAt("spec")
+	verr.AddErrorAt(path.Field("targetRef"), validateTop(r.Spec.TargetRef))
+	verr.AddErrorAt(path.Field("default"), validateDefault(r.Spec.Default))
+	return verr.OrNil()
+}
+
+func validateTop(targetRef common_api.TargetRef) validators.ValidationError {
+	targetRefErr := mesh.ValidateTargetRef(targetRef, &mesh.ValidateTargetRefOpts{
+		SupportedKinds: []common_api.TargetRefKind{
+			common_api.Mesh,
+			common_api.MeshSubset,
+		},
+	})
+	return targetRefErr
+}
+
+func validateDefault(conf Conf) validators.ValidationError {
+	var verr validators.ValidationError
+	portAndProtocol := map[int]ProtocolType{}
+	for i, match := range conf.AppendMatch {
+		if match.Port != nil && pointer.Deref[int](match.Port) == 0 || pointer.Deref[int](match.Port) > math.MaxUint16 {
+			verr.AddViolationAt(validators.RootedAt("appendMatch").Index(i).Field("port"), "port must be a valid (1-65535)")
+		}
+		if match.Port != nil {
+			if value, found := portAndProtocol[pointer.Deref[int](match.Port)]; found && value != match.Protocol && slices.Contains(notAllowedProtocolsOnTheSamePort, match.Protocol) {
+				verr.AddViolationAt(validators.RootedAt("appendMatch").Index(i).Field("port"), fmt.Sprintf("protocol needs to be the same across different port when using: %v", notAllowedProtocolsOnTheSamePort))
```

Comment:
What is the requirement here? Is it that using the same port in multiple matches requires the same protocol? If so, then should this be:

```suggestion
				verr.AddViolationAt(validators.RootedAt("appendMatch").Index(i).Field("port"), fmt.Sprintf("using the same port in multiple matches requires the same protocol for the following protocols: %v", notAllowedProtocolsOnTheSamePort))
```

maybe we should drop the match to "tcp" and have a warning?

Reply:
The requirement here: when there are many different protocols on the same port check if we can distinguish the traffic. 
* When there is TCP/TLS  and (one of http/http2/grpc), it's fine we can figure it out.
* When there is 2 of HTTP/HTTP2/GRPC we cannot distinguish traffic easily